### PR TITLE
Add in Okta plugin type.

### PR DESCRIPTION
**Pull request by mdwn**
The Okta plugin type has been added to support the hosted Okta plugin in enterprise.

### CODE REVIEW

**Review by probakowski — APPROVED** [review]

File: api/types/plugin.go [hunk 1]
```diff
@@ -111,6 +113,25 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if (bearer.Token == "") == (bearer.TokenFile == "") {
 			return trace.BadParameter("exactly one of Token and TokenFile must be specified")
 		}
+	case *PluginSpecV1_Okta:
+		// Check settings.
+		if settings.Okta == nil {
+			return trace.BadParameter("settings must be set")
```
Comment (GavinFrazar) [inline api/types/plugin.go:119]:
```suggestion
			return trace.BadParameter("missing Okta settings")
```
nit: I think adding a little more info to the error would be helpful

File: api/types/plugin.go [hunk 2]
```diff
@@ -257,6 +280,15 @@ func (s *PluginSlackAccessSettings) CheckAndSetDefaults() error {
 	return nil
 }
 
+// CheckAndSetDefaults validates and set the default values
```
Comment (GavinFrazar) [inline api/types/plugin.go:283]:
```suggestion
// CheckAndSetDefaults validates and set the default values.
```
nit

File: api/types/plugin.go [hunk 3]
```diff
@@ -111,6 +113,25 @@ func (p *PluginV1) CheckAndSetDefaults() error {
...
+		}
+
+		if p.Credentials == nil {
+			return trace.BadParameter("credentials must be set")
+		}
+		bearer := p.Credentials.GetBearerToken()
+		if bearer == nil {
+			return trace.BadParameter("okta plugin must be used with the bearer token credential type")
+		}
+		if (bearer.Token == "") == (bearer.TokenFile == "") {
```
Comment (GavinFrazar) [inline api/types/plugin.go:132]: huh, that's a clever way to do it 👍

**Review by GavinFrazar — APPROVED** [review]

File: api/types/plugin.go (hunk 3 above)
Comment (mdwn) [reply api/types/plugin.go:132]: I can't take credit for it. :-) Copy paste job!

**Pull request merged**